src/test/java/org/fogbowcloud/manager/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -90,577 +84,847 @@
         this.volumePlugin = Mockito.mock(VolumePlugin.class);
         this.imagePlugin = Mockito.mock(ImagePlugin.class);
         this.computeQuotaPlugin = Mockito.mock(ComputeQuotaPlugin.class);
         
+        // mocking instances/image and the return of getID method
         this.networkInstance = Mockito.mock(NetworkInstance.class);
-        when(networkInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
+        Mockito.when(networkInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
         
         this.volumeInstance = Mockito.mock(VolumeInstance.class);
-        when(volumeInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
+        Mockito.when(volumeInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
         
         this.attachmentInstance = Mockito.mock(AttachmentInstance.class);
-        when(attachmentInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
+        Mockito.when(attachmentInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
         
         this.computeInstance = Mockito.mock(ComputeInstance.class);
-        when(computeInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
+        Mockito.when(computeInstance.getId()).thenReturn(FAKE_INSTANCE_ID);
        
         this.image = Mockito.mock(Image.class);
-        when(image.getId()).thenReturn(FAKE_IMAGE_ID);
+        Mockito.when(image.getId()).thenReturn(FAKE_IMAGE_ID);
         
-        when(cloudPluginsHolder.getComputePlugin()).thenReturn(computePlugin);
-        when(cloudPluginsHolder.getAttachmentPlugin()).thenReturn(attachmentPlugin);
-        when(cloudPluginsHolder.getNetworkPlugin()).thenReturn(networkPlugin);
-        when(cloudPluginsHolder.getVolumePlugin()).thenReturn(volumePlugin);
-        when(cloudPluginsHolder.getImagePlugin()).thenReturn(imagePlugin);
-        when(cloudPluginsHolder.getComputeQuotaPlugin()).thenReturn(computeQuotaPlugin);
-        when(aaController.getLocalToken(any(FederationUser.class))).thenReturn(new Token());
+        // mocking cloudPluginsHolder to return the correct plugin for each call
+        Mockito.when(cloudPluginsHolder.getComputePlugin()).thenReturn(computePlugin);
+        Mockito.when(cloudPluginsHolder.getAttachmentPlugin()).thenReturn(attachmentPlugin);
+        Mockito.when(cloudPluginsHolder.getNetworkPlugin()).thenReturn(networkPlugin);
+        Mockito.when(cloudPluginsHolder.getVolumePlugin()).thenReturn(volumePlugin);
+        Mockito.when(cloudPluginsHolder.getImagePlugin()).thenReturn(imagePlugin);
+        Mockito.when(cloudPluginsHolder.getComputeQuotaPlugin()).thenReturn(computeQuotaPlugin);
+        Mockito.when(aaController.getLocalToken(Mockito.any(FederationUser.class))).thenReturn(new Token());
         
+        // starting the object we want to test
         this.localCloudConnector = new LocalCloudConnector(aaController, cloudPluginsHolder);
         
     }
 	
+	// test case: Request a compute instance Mockito.when the plugin returns a correct id
```

Comment:
I did not understand the "Mockit.when" in this comment.

Reply:
I've fixed it.